cpp/src/arrow/dataset/file_ipc.cc

```diff
@@ -121,6 +137,10 @@ Result<std::shared_ptr<Schema>> IpcFileFormat::Inspect(const FileSource& source)
 Result<RecordBatchGenerator> IpcFileFormat::ScanBatchesAsync(
     const std::shared_ptr<ScanOptions>& options,
     const std::shared_ptr<FileFragment>& file) const {
+#ifdef ARROW_WITH_OPENTELEMETRY
+  auto tracer = arrow::internal::tracing::GetTracer();
+  auto parent_span = tracer->GetCurrentSpan();
+#endif
```

Comment:
Does this actually have an effect anymore?

Reply:
You're right, they are left-overs... I've removed them, thanks